src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -26,34 +24,55 @@ internal partial class DiagnosticIncrementalAnalyzer
         private async Task<ProjectAnalysisData> GetProjectAnalysisDataAsync(
             CompilationWithAnalyzers? compilationWithAnalyzers, Project project, IdeAnalyzerOptions ideOptions, ImmutableArray<StateSet> stateSets, CancellationToken cancellationToken)
         {
-            using (Logger.LogBlock(FunctionId.Diagnostics_ProjectDiagnostic, GetProjectLogMessage, project, stateSets, cancellationToken))
+            // Compute the diagnostic data for all the requested state sets for this particular project.
+            var projectAnalysisData = await GetProjectAnalysisDataWorkerAsync().ConfigureAwait(false);
+
+            // Now, save all that data to the in-memory storage, so it can be retrieved later when making requests for
+            // other documents within the same project.
+            await SaveAllStatesToInMemoryStorageAsync().ConfigureAwait(false);
```

Comment:
@mavasani i was hoping this would help.  By making it so that we always save for any caller of `GetProjectAnalysisDataAsync` (currently there are two of htem).

However, it looks like this doesn't help because the code that attempts to load from teh cache seems to bail out for any number of reasons i can't make heads or tails of.